libethashseal/Ethash.cpp

```diff
@@ -233,21 +235,15 @@ void Ethash::populateFromParent(BlockHeader& _bi, BlockHeader const& _parent) co
     _bi.setGasLimit(childGasLimit(_parent));
 }
 
-bool Ethash::quickVerifySeal(BlockHeader const& _bi) const
+bool Ethash::quickVerifySeal(BlockHeader const& _blockHeader) const
 {
-    if (_bi.number() >= ETHASH_EPOCH_LENGTH * 2048)
-        return false;
+    h256 const h = _blockHeader.hash(WithoutSeal);
+    h256 const b = boundary(_blockHeader);
+    uint64_t const n = (uint64_t)(u64)nonce(_blockHeader);
```

Comment:
Is this double cast necessary?

Reply:
Yes, because `Nonce` is `h64` and there is no conversion from `Nonce` to `uint64_t`.